### PR TITLE
arm64: dts: qcom: msm8939-alcatel-idol3: enable smb1360

### PR DESCRIPTION
**Pull request by vknecht**

### CODE REVIEW

File: arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts [hunk 1]
```diff
@@ -0,0 +1,773 @@
...
+#include <dt-bindings/iio/qcom,spmi-vadc.h>
+#include <dt-bindings/leds/common.h>
+#include <dt-bindings/pinctrl/qcom,pmic-mpp.h>
+
+/ {
+	model = "Alcatel OneTouch Idol 3 (5.5)";
+	compatible = "alcatel,idol3", "qcom,msm8939-mtp", "qcom,msm8939", "qcom,mtp";
+	chassis-type = "handset";
+	qcom,msm-id = <239 0>;
+	qcom,board-id = <8 0>;
```
Comment (stephan-gh) [inline arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts]: Drop these please

File: arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts [hunk 2]
```diff
@@ -0,0 +1,773 @@
...
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/iio/qcom,spmi-vadc.h>
+#include <dt-bindings/leds/common.h>
+#include <dt-bindings/pinctrl/qcom,pmic-mpp.h>
+
+/ {
+	model = "Alcatel OneTouch Idol 3 (5.5)";
+	compatible = "alcatel,idol3", "qcom,msm8939-mtp", "qcom,msm8939", "qcom,mtp";
```
Comment (stephan-gh) [inline arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts]: FIXME

File: arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts [hunk 3]
```diff
@@ -0,0 +1,773 @@
...
+	};
+
+	reserved-memory {
+		/delete-node/ venus@8c300000;
+
+		cont_splash_mem: framebuffer@83000000 {
+			compatible = "removed-dma-pool";
+			reg = <0x0 0x83000000 0x0 0x2000000>;
+			no-map;
+		};
```
Comment (stephan-gh) [inline arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts]: Is it worth reserving this? I think you just waste memory since it's not needed anymore after the display driver runs

File: arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts [hunk 4]
```diff
@@ -0,0 +1,773 @@
...
+	aliases {
+		serial0 = &blsp1_uart2;
+	};
+
+	chosen {
+		stdout-path = "serial0";
+	};
+
+	reserved-memory {
+		/delete-node/ venus@8c300000;
```
Comment (stephan-gh) [inline arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts]: Why is this needed?

File: arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts [hunk 5]
```diff
@@ -0,0 +1,773 @@
...
+			reg = <0x0 0x8c400000 0x0 0x200000>;
+			no-map;
+		};
+
+		venus_mem: venus@90000000 {
+			reg = <0x0 0x90000000 0x0 0x800000>;
+			no-map;
+		};
+
+		ramoops: ramoops_mem_region@9ff00000 {
```
Comment (stephan-gh) [inline arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts]: `ramoops@`

File: arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts [hunk 6]
```diff
@@ -0,0 +1,773 @@
...
+			compatible = "ramoops";
+			reg = <0 0x9ff00000 0 0x100000>;
+
+			record-size = <0x1000>;
+			console-size = <0x40000>;
+			ftrace-size = <0x1000>;
+			//pmsg-size = <0x20000>;
```
Comment (stephan-gh) [inline arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts]: Please decide if you want this or not. But are you actually using ramoops? Does it work properly?

File: arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts [hunk 7]
```diff
@@ -0,0 +1,773 @@
...
+
+	backlight: backlight {
+		compatible = "pwm-backlight";
+		pwms = <&pm8916_pwm 0 255000>;
+		brightness-levels = <0 255>;
+		num-interpolated-steps = <255>;
+		default-brightness-level = <64>;
+	};
+
+	gpio-keys {
```
Comment (stephan-gh) [inline arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts]: I'd add this exactly like in alcatel-idol347, there are some unneeded things in here

File: arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts [hunk 8]
```diff
@@ -0,0 +1,773 @@
...
+			linux,can-disable;
+		};
+	};
+
+	reg_lcd_enp: regulator-lcd-enp {
+		compatible = "regulator-fixed";
+		regulator-name = "lcd_enp";
+		regulator-min-microvolt = <1800000>;
+		regulator-max-microvolt = <1800000>;
+		regulator-boot-on; // allows fb console to show up briefly at boot without always-on
```
Comment (stephan-gh) [inline arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts]: I think the comment is unneeded, but if you want it please use the `/* */` comment format

File: arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts [hunk 9]
```diff
@@ -0,0 +1,773 @@
...
+		pinctrl-0 = <&accel_int_default>;
+		mount-matrix = "-1", "0", "0",
+			       "0", "-1", "0",
+			       "0", "0", "-1";
+	};
+
+	proximity@39 {
+		compatible = "amstaos,tsl2772";
+		reg = <0x39>;
+		interrupts-extended = <&msmgpio 34 IRQ_TYPE_EDGE_FALLING>;
```
Comment (stephan-gh) [inline arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts]: It's fine but it's strange that you are using interrupts-extended only here and interrupt-parent+interrupts otherwise

File: arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts [hunk 10]
```diff
@@ -0,0 +1,773 @@
...
+&blsp_i2c2 {
+	status = "okay";
+
+	magnetometer@c {
+		compatible = "asahi-kasei,ak09911";
+		reg = <0x0c>;
+		vdd-supply = <&pm8916_l17>;
+		vid-supply = <&pm8916_l6>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&mag_reset_default>;
```
Comment (stephan-gh) [inline arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts]: should you add the reset-gpios here?

File: arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts [hunk 11]
```diff
@@ -0,0 +1,773 @@
...
+		pinctrl-1 = <&proximity_sleep>;
+	};
+
+	gyroscope@68 {
+		compatible = "bosch,bmg160";
+		reg = <0x68>;
+		vdd-supply = <&pm8916_l17>;
+		vddio-supply = <&pm8916_l6>;
+		interrupt-parent = <&msmgpio>;
+		interrupts = <23 IRQ_TYPE_EDGE_RISING>;
```
Comment (stephan-gh) [inline arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts]: please add pinctrl for this

File: arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts [hunk 12]
```diff
@@ -0,0 +1,773 @@
...
+		reg = <0x10>;
+		tvdd-supply = <&pm8916_l6>;
+		pdn-gpios = <&msmgpio 101 GPIO_ACTIVE_HIGH>;
+		pinctrl-names = "default";
+		pintctrl-0 = <&headphones_pdn_default>;
+		#sound-dai-cells = <0>;
+	};
+
+	charger: battery@14 {
+		compatible = "qcom,smb1360";
```
Comment (stephan-gh) [inline arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts:192]: Please make a separate commit for this since that driver is not upstream yet

File: arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts [hunk 13]
```diff
@@ -0,0 +1,773 @@
...
+		qcom,shdn-after-pwroff;
+		qcom,fg-reset-at-pon;
+		qcom,fg-reset-threshold-mv = <200>;
+
+		usb_otg_vbus: usb-otg-vbus {
+			regulator-max-microamp = <1500000>;
+		};
+	};
+
+	ncp6335d: ncp6335d-regulator@1c {
```
Comment (stephan-gh) [inline arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts]: I'd drop this completely for now, please just leave the CPU regulator the way it is

File: arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts [hunk 14]
```diff
@@ -0,0 +1,773 @@
...
+	};
+
+	ports {
+		port@1 {
+			endpoint {
+				remote-endpoint = <&panel_in>;
+				data-lanes = <1 2 3 0>;
+			};
+		};
+	};
```
Comment (stephan-gh) [inline arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts]: Please set this up like in idol347 (with `&dsi0_out`)

File: arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts [hunk 15]
```diff
@@ -0,0 +1,773 @@
...
+};
+
+&dsi_phy0 {
+	status = "okay";
+	vddio-supply = <&pm8916_l6>;
+	qcom,dsi-phy-regulator-ldo-mode;
+};
+
+&lpass {
```
Comment (stephan-gh) [inline arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts]: shouldn't be here if you have modem enabled

File: arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts [hunk 16]
```diff
@@ -0,0 +1,773 @@
...
+
+&sdhc_2 {
+	status = "okay";
+
+	vmmc-supply = <&pm8916_l11>;
+	vqmmc-supply = <&pm8916_l12>;
+
+	pinctrl-names = "default", "sleep";
+	pinctrl-0 = <&sdc2_clk_on &sdc2_cmd_on &sdc2_data_on>;
+	pinctrl-1 = <&sdc2_clk_off &sdc2_cmd_off &sdc2_data_off>;
```
Comment (stephan-gh) [inline arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts]: cd-gpios?

File: arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts [hunk 17]
```diff
@@ -0,0 +1,773 @@
...
+			regulator-min-microvolt = <1200000>;
+			regulator-max-microvolt = <1300000>;
+		};
+
+		pm8916_s4: s4 {
+			regulator-min-microvolt = <1800000>;
+			regulator-max-microvolt = <2100000>;
+		};
+
+		/* l1 is fixed to 1225000, but not connected in schematic */
```
Comment (stephan-gh) [inline arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts]: you have one (a schematic)? otherwise this seems like a bad copy/paste

File: arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts [hunk 18]
```diff
@@ -0,0 +1,773 @@
...
+		pm8916_l5: l5 {
+			regulator-min-microvolt = <1800000>;
+			regulator-max-microvolt = <1800000>;
+			regulator-always-on; /* keep it ? */
+		};
+
+		pm8916_l6: l6 {
+			regulator-min-microvolt = <1800000>;
+			regulator-max-microvolt = <1800000>;
+			regulator-always-on;
```
Comment (stephan-gh) [inline arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts]: Is this needed?

File: arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts [hunk 19]
```diff
@@ -0,0 +1,773 @@
...
+
+		pm8916_l4: l4 {
+			regulator-min-microvolt = <2050000>;
+			regulator-max-microvolt = <2050000>;
+		};
+
+		pm8916_l5: l5 {
+			regulator-min-microvolt = <1800000>;
+			regulator-max-microvolt = <1800000>;
+			regulator-always-on; /* keep it ? */
```
Comment (stephan-gh) [inline arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts]: At least on 8916 l5 is always-on in hardware iirc so it's probably fine to keep it but also shouldn't be strictly needed

File: arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts [hunk 20]
```diff
@@ -84,6 +84,16 @@
 		};
 	};
 
+	gpio-leds {
+		compatible = "gpio-leds";
+
+		led-0 {
+			gpios = <&msmgpio 32 GPIO_ACTIVE_HIGH>;
```
Comment (stephan-gh) [inline arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts]: Please add pinctrl. And I'd squash this with the initial commit

File: arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts [hunk 21]
```diff
@@ -132,6 +132,18 @@
 		pinctrl-0 = <&lcd_enn_default>;
 	};
 
+	reg_led_enable: regulator-led-enable {
+		compatible = "regulator-fixed";
+		regulator-name = "led_enable";
+		regulator-min-microvolt = <1800000>;
+		regulator-max-microvolt = <1800000>;
+		regulator-always-on;
```
Comment (stephan-gh) [inline arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts]: Why does this need to be always-on?

File: arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts (hunk 6 above)
Comment (vknecht) [reply arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts]: It works to some point, doesn't seem too reliable (not always available when trying to view it in twrp)

File: arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts (hunk 18 above)
Comment (vknecht) [reply arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts]: Seems I've added it so that panel works, it doesn't display anything without it (only backlight lit)

Comment (stephan-gh) [reply arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts]: Hm, but l6 is specified as supply of both DSI and panel, so it shouldn't get turned off. :confused:

Comment (vknecht) [reply arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts]: Scratch that, it just that I forgot to generate panels again after cleaning the branch for pushing here...

File: arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts (hunk 4 above)
Comment (vknecht) [reply arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts]: This is for letting gps@8c400000 live... would it be better to move venus in dtsi ?

Comment (stephan-gh) [reply arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts]: gps_mem can mostly be anywhere (ideally in a contiguous chunk of reserved memory), you could also just put it after mba_mem

File: arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts (hunk 21 above)
Comment (vknecht) [reply arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts]: This is needed because OEM added a second "enable" LED GPIO which is not in IC datasheet nor supported by the driver, so that it's always enabled and LED driver can drive the IC with the only documented enable GPIO...
Without always-on, the LED can't be lit up again once the driver loads and shuts it down (LED is on at boot).

Comment (stephan-gh) [reply arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts]: So the GPIO is a bit like "hand over control of LED to IC thingy"?

If it's not really a regulator and you just want this GPIO to be high permanently then it's easier and cleaner to make a pinctrl entry for it with `output-high`. You can add this pinctrl to the `sn3190`.

Comment (stephan-gh) [reply arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts]: A comment on the pinctrl with something like your explanation above would be nice too :)

Comment (stephan-gh) [reply arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts]: BTW, does idol347 have the same LED thing?

Comment (vknecht) [reply arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts]: yes, idol347 has it too, as well as gpio torch led, will add them later

Comment (vknecht) [reply arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts]: Done. Must I say that I don't know if it's really gpio9 which is wired on shutdown-gpios, and not the gpio108 ? :-)

File: arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts [hunk 22]
```diff
@@ -0,0 +1,796 @@
...
+	 * the si-en,sn3190 LED IC works. Since this GPIO is not
+	 * part of the IC datasheet nor supported by the driver,
+	 * force it asserted here.
+	 */
+	led_enable_default: led-enable-default {
+		pins = "gpio108";
+		function = "gpio";
+
+		drive-strength = <2>;
+		bias-pull-up;
```
Comment (stephan-gh) [inline arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts]: please change it here as well

File: arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts [hunk 23]
```diff
@@ -0,0 +1,703 @@
...
+		compatible = "amstaos,tsl2772";
+		reg = <0x39>;
+		interrupts-extended = <&msmgpio 34 IRQ_TYPE_EDGE_FALLING>;
+		vdd-supply = <&pm8916_l17>;
+		vddio-supply = <&pm8916_l6>;
+		amstaos,proximity-diodes = <0>;
+		led-max-microamp = <100000>;
+		pinctrl-names = "default", "sleep";
+		pinctrl-0 = <&proximity_default>;
+		pinctrl-1 = <&proximity_sleep>;
```
Comment (stephan-gh) [inline arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts]: The tsl2772 driver doesn't pick the sleep pinctrl state so you might as well omit it.

File: arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts [hunk 24]
```diff
@@ -0,0 +1,703 @@
...
+		pinctrl-0 = <&speaker_bottom_default>;
+		sound-name-prefix = "Speaker Bottom";
+		#sound-dai-cells = <0>;
+	};
+};
+
+&blsp_i2c5 {
+	status = "okay";
+
+	synaptics@20 {
```
Comment (stephan-gh) [inline arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts]: this is rmi4@ usually I think

File: arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts [hunk 25]
```diff
@@ -0,0 +1,703 @@
...
+		status = "disabled";
+		compatible = "syna,rmi4-i2c";
+		reg = <0x20>;
+		interrupts-extended = <&msmgpio 13 IRQ_TYPE_LEVEL_LOW>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&ts_int_active &ts_reset_active>;
+		#address-cells = <1>;
+		#size-cells = <0>;
```
Comment (stephan-gh) [inline arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts]: vdd/vio supply?

File: arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts [hunk 26]
```diff
@@ -0,0 +1,703 @@
...
+
+		rmi4-f11@11 {
+			reg = <0x11>;
+			syna,sensor-type = <1>;
+		};
+	};
+
+	touchscreen@38 {
+		status = "okay";
+		compatible = "edt,edt-ft5406", "edt,edt-ft5x06";
```
Comment (stephan-gh) [inline arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts]: `edt,edt-ft5x06` compatible isn't documented, is the first enough?

File: arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts [hunk 27]
```diff
@@ -0,0 +1,703 @@
...
+		};
+
+		rmi4-f11@11 {
+			reg = <0x11>;
+			syna,sensor-type = <1>;
+		};
+	};
+
+	touchscreen@38 {
+		status = "okay";
```
Comment (stephan-gh) [inline arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts]: I'd disable both touchscreens by default, we'll need to enable one of them in lk2nd.

File: arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts [hunk 28]
```diff
@@ -0,0 +1,703 @@
...
+	pinctrl-names = "default", "sleep";
+	pinctrl-0 = <&mdss_default>;
+	pinctrl-1 = <&mdss_sleep>;
+
+	panel@0 {
+		compatible = "alcatel,idol3-panel";
+		reg = <0>;
+
+		reset-gpios = <&msmgpio 25 GPIO_ACTIVE_LOW>;
+		power-supply = <&pm8916_l17>;
```
Comment (stephan-gh) [inline arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts]: I wonder if l17 is really used here or if the lcd_enp/enn are enough (similar to idol347)

File: arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts [hunk 29]
```diff
@@ -0,0 +1,703 @@
...
+	ts_pins: ts-pins {
+		ts_int_active: pmx-ts-int-active {
+			pins = "gpio13";
+			function = "gpio";
+
+			drive-strength = <16>;
+			bias-pull-up;
+		};
+
+		ts_reset_active: pmx-ts-reset-active {
```
Comment (stephan-gh) [inline arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts]: If both ts-pins are configured the same you could just put them into one node. But I don't quite see the point why the reset gpio is configured with bias-pull-up, it should be fine with bias-disable. Also, the drive-strength seems a bit excessive...

File: arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts [hunk 30]
```diff
@@ -194,6 +194,50 @@
...
+	charger: battery@14 {
+		compatible = "qcom,smb1360";
+		reg = <0x14>;
+
+		interrupts-extended = <&msmgpio 62 IRQ_TYPE_LEVEL_LOW>;
+
+		pinctrl-names = "default";
+		pinctrl-0 = <&smb_int_default>;
+
+		status = "okay";
```
Comment (stephan-gh) [inline arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts]: Can drop this line

File: arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts (hunk 29 above)
Comment (vknecht) [reply arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts]: Seems fine with both drive-strength = 2.
Now the two states only differ about bias, not sure how to factor that in one node ?

File: arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts (hunk 16 above)
Comment (stephan-gh) [reply arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts]: should also have pinctrl for the cd-gpios though

File: arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts (hunk 29 above)
Comment (stephan-gh) [reply arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts]: It's fine now, please drop the outer ts-pins node though.

File: arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts (hunk 27 above)
Comment (vknecht) [reply arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts]: Done, looking forward to this lk2nd feature :-)

File: arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts [hunk 31]
```diff
@@ -0,0 +1,697 @@
...
+	};
+
+	gpio-lid-sensor {
+		compatible = "gpio-keys";
+
+		label = "GPIO Lid Sensor";
+
+		lid-sensor {
+			label = "Lid Sensor";
+			gpios = <&msmgpio 98 GPIO_ACTIVE_LOW>;
```
Comment (stephan-gh) [inline arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts]: pinctrl? It's good practice to add it for everything :p

File: arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts [hunk 32]
```diff
@@ -0,0 +1,697 @@
...
+
+	proximity_default: proximity-default-state {
+		pins = "gpio34";
+		function = "gpio";
+
+		drive-strength = <6>;
+		bias-pull-up;
+	};
+
+	pmx_mdss {
```
Comment (stephan-gh) [inline arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts]: guess we should drop this outer node as well: https://git.kernel.org/pub/scm/linux/kernel/git/qcom/linux.git/commit/?h=for-next&id=8b276ca036377a5baa4fd0692b80608f0de8a260

File: arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts [hunk 33]
```diff
@@ -214,6 +214,23 @@
 		sound-name-prefix = "Speaker Bottom";
 		#sound-dai-cells = <0>;
 	};
+
+	led-controller@68 {
```
Comment (stephan-gh) [inline arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts]: I'd squash this with the first commit, or is this not upstream yet?

File: arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts [hunk 34]
```diff
@@ -193,6 +193,48 @@
 		#sound-dai-cells = <0>;
 	};
 
+	charger: battery@14 {
```
Comment (stephan-gh) [inline arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts:191]: Are you absolutely sure about this? So far we usually verified all the configurations by comparing the dumps with the downstream driver. I know it's quite some effort but I don't want to be responsible for a wrong charging configuration.

File: arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts (hunk 31 above)
Comment (vknecht) [reply arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts]: Removed the lid sensor for now, it's been a while since I added and tested it and now it cuts at least USB connectivity

File: arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts (hunk 34 above)
Comment (vknecht) [reply arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts:191]: Not absolutely sure, no. Even tho it seems fine since quite a while, there's at least a little glitch in that even when gauge is at 100%, the charging still stays on for a while according to phosh indicator.
Should it stop at 95% iiuc `qcom,fg-auto-recharge-soc = <95>;` ?

Comment (stephan-gh) [reply arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts:191]: Is the "glitch" any different on idol347?

> Should it stop at 95% iiuc qcom,fg-auto-recharge-soc = <95>; ?

I think it generally means that it would charge to 100%, then wait until 95% before it charges again. But I'm not sure how that is exposed to the user, i.e. if you would see 100% the whole time.

Comment (vknecht) [reply arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts:191]: same on idol347...

**Comment by stephan-gh** [comment]
Can you update to the `msm8916/6.1-rc5` branch? I added the `msm8939-msm8916.dtsi` include similar to the `msm8916-pm8916.dtsi` used on all the MSM8916 devices, so please:

- `#include "msm8939-pm8916.dtsi"` instead of `#include "msm8939.dtsi"` and `#include "pm8916.dtsi"`
- Drop all the `-supply`s that are already in `msm8939-pm8916.dtsi`
- Change the regulator definitions to be like in MSM8916 devices:
  - `&rpm_requests` -> `&smd_rpm_regulators` (drop the extra subnode)
  - Optional: Can drop the label from all regulator nodes (e.g. `pm8916_s3: s3 { ... }` -> `s3 { ... };`
- See also: https://wiki.postmarketos.org/wiki/MSM8916_Mainlining#Regulators

File: arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts [hunk 35]
```diff
@@ -0,0 +1,756 @@
...
+	pinctrl-names = "default", "sleep";
+	pinctrl-0 = <&mdss_default>;
+	pinctrl-1 = <&mdss_sleep>;
+
+	panel@0 {
+		compatible = "alcatel,idol3-panel";
+		reg = <0>;
+
+		reset-gpios = <&msmgpio 25 GPIO_ACTIVE_LOW>;
+		vddio-supply = <&pm8916_l6>;
```
Comment (vknecht) [inline arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts]: guess I could wipe this vddio-supply reference too ?

Comment (stephan-gh) [reply arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts]: probably yeah, it should already come through the dsi port.

File: arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts [hunk 36]
```diff
@@ -0,0 +1,756 @@
...
+		led@1 {
+			reg = <1>;
+			led-max-microamp = <5000>;
+			function = LED_FUNCTION_INDICATOR;
+			color = <LED_COLOR_ID_WHITE>;
+		};
+	};
+};
+
+&blsp_i2c5 {
```
Comment (stephan-gh) [inline arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts]: don't think lk2nd will also enable the i2c bus, so I think we need a status = "okay" here :)

File: arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts [hunk 37]
```diff
@@ -0,0 +1,757 @@
...
+		id-gpio = <&msmgpio 69 GPIO_ACTIVE_HIGH>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&usb_id_default>;
+	};
+};
+
+&blsp_dma {
+	status = "okay";
+};
+
```
Comment (wonderfulShrineMaidenOfParadise) [inline arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts]: Please try #256, drop this node and see if i2c1~6 still works.

File: arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts [hunk 38]
```diff
@@ -0,0 +1,706 @@
...
+	gpio-leds {
+		compatible = "gpio-leds";
+
+		pinctrl-names = "default";
+		pinctrl-0 = <&gpio_leds_default>;
+
+		led-0 {
+			gpios = <&msmgpio 32 GPIO_ACTIVE_HIGH>;
+			linux,default-trigger = "torch";
```
Comment (stephan-gh) [inline arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts]: Is `torch` really valid and doing anything here? This is not documented in the DT schema: https://github.com/torvalds/linux/blob/aae703b02f92bde9264366c545e87cec451de471/Documentation/devicetree/bindings/leds/common.yaml#L76-L102

File: arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts [hunk 39]
```diff
@@ -0,0 +1,706 @@
...
+	pinctrl-names = "default", "sleep";
+	pinctrl-0 = <&sdc1_clk_on &sdc1_cmd_on &sdc1_data_on>;
+	pinctrl-1 = <&sdc1_clk_off &sdc1_cmd_off &sdc1_data_off>;
+};
+
+&sdhc_2 {
+	status = "okay";
+	pinctrl-names = "default", "sleep";
+	pinctrl-0 = <&sdc2_clk_on &sdc2_cmd_on &sdc2_data_on &sdhc2_cd_default>;
+	pinctrl-1 = <&sdc2_clk_off &sdc2_cmd_off &sdc2_data_off &sdhc2_cd_default>;
```
Comment (stephan-gh) [inline arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts]: There is `&sdc2_cd_on`/off predefined for the cd-gpios which configures the pull-up that is probably necessary.

File: arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts [hunk 40]
```diff
@@ -188,6 +188,48 @@
...
+		qcom,float-voltage-mv = <4350>;
+		qcom,iterm-ma = <100>;
+		qcom,charging-timeout = <192>;
+		qcom,recharge-thresh-mv = <100>;
+		qcom,chg-inhibit-disabled;
+
+		qcom,battery-profile = <1>; /* downstream forces profile B */
+		qcom,fg-batt-capacity-mah = <2910>;
+		qcom,fg-cc-soc-coeff = <0x81A1>;
+		qcom,fg-cutoff-voltage-mv = <3400>;
```
Comment (stephan-gh) [inline arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts]: Is this accidentally copied from idol347? I don't see this on downstream

Comment (vknecht) [reply arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts]: guess so, I removed it

File: arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts [hunk 41]
```diff
@@ -188,6 +188,49 @@
...
+		qcom,warm-bat-decidegc = <450>;
+		qcom,cool-bat-decidegc = <100>;
+		qcom,soft-jeita-comp-voltage-mv = <4100>;
+		qcom,soft-jeita-comp-current-ma = <750>;
+
+		qcom,shdn-after-pwroff;
+		qcom,fg-reset-at-pon;
+		qcom,fg-reset-threshold-mv = <200>;
+
+		status = "okay";
```
Comment (stephan-gh) [inline arch/arm64/boot/dts/qcom/msm8939-alcatel-idol3.dts]: FWIW you can drop this since no `status` at all is equivalent to `okay`

**Review by stephan-gh — APPROVED** [review]

**Comment by stephan-gh** [comment]
Cherry-picked to `wip/msm8916/6.1`. Thanks!
